### PR TITLE
Enhance re share tests

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
1) Adjust test scenarios so that `user0` has skeleton files, and other users (who receive shares) do not have skeleton files. This makes the test sequences a bit easier, because the received share names do not have `(2)` on them.

2) Add some extra `Then` steps to existing scenarios to check that the correct share and reshare recipients can or cannot see the shared file/folder. This helps verify that the sharing and resharing combinations really are working, and not just that some OCS/HTTP status codes were as expected.

3) Add more combinations of resharing with less, the same and more permissions of the received share, and of a sub-folder in a received share.

4) Add scenarios for resharing and then (attempting to) update the re-share to have more/same/less permissions.

5) Add scenarios for creating a public link share of a received share. Make sure that you can't give more permissions in the public link share then what you received from the original sharer.

6) Added 2 scenarios for issue #35528 related to trying to increase the permissions on a reshare, where you are just increasing the permissions to a level that should be allowed.

7) Added various combinations of steps for "the public downloads file" with a without password, and "the public should be able to download file" etc that were missing. There were already steps that did this in some combinations, and checking ranges of bytes in the file etc. 

## Related Issue
https://github.com/owncloud/enterprise/issues/3364

## How Has This Been Tested?
Local acceptance test runs

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiShareManagement/reShare.feature [hunk 1]
```diff
@@ -99,38 +309,167 @@ Feature: sharing
...
+    And user "user0" has created folder "/test"
+    And user "user0" has shared folder "/test" with user "user1" with permissions 17
+    #And user "user1" has created folder "/qqq"
+    When user "user1" creates a public link share using the sharing API with settings
+      | path         | /test |
+      | permissions  | 15    |
+      | publicUpload | true  |
+    Then the OCS status code should be "404"
+    And the HTTP status code should be "<http_status_code>"
+    #And publicly uploading a file should work
```
Comment (individual-it) [inline tests/acceptance/features/apiShareManagement/reShare.feature]: is there an issue with that scenario or why that commented line?

File: tests/acceptance/features/apiShareManagement/reShare.feature [hunk 2]
```diff
@@ -99,38 +309,167 @@ Feature: sharing
...
+    And user "user0" has created folder "/test"
+    And user "user0" has created folder "/test/sub"
+    And user "user0" has shared folder "/test" with user "user1" with permissions 17
+    When user "user1" creates a public link share using the sharing API with settings
+      | path         | /test/sub |
+      | permissions  | 15        |
+      | publicUpload | true      |
     Then the OCS status code should be "404"
     And the HTTP status code should be "<http_status_code>"
+    #And publicly uploading a file should work
```
Comment (individual-it) [inline tests/acceptance/features/apiShareManagement/reShare.feature]: why this two lines? there is no public link

**Review by individual-it — CHANGES_REQUESTED** [review]
some minor requests

File: tests/acceptance/features/apiShareManagement/reShare.feature [hunk 3]
```diff
@@ -99,38 +309,167 @@ Feature: sharing
...
+      | ocs_api_version | ocs_status_code |
+      | 1               | 100             |
+      | 2               | 200             |
+
+  @public_link_share-feature-required
+  Scenario Outline: creating an upload public link from a share with share+read only permissions is not allowed
+    Given using OCS API version "<ocs_api_version>"
+    And user "user0" has created folder "/test"
+    And user "user0" has shared folder "/test" with user "user1" with permissions 17
+    #And user "user1" has created folder "/qqq"
```
[the remark this answers is not in the document]
Comment (phil-davis) [reply tests/acceptance/features/apiShareManagement/reShare.feature]: My mistake - I had been debugging my scenario at one point.

File: tests/acceptance/features/apiShareManagement/reShare.feature (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/apiShareManagement/reShare.feature]: mistake

File: tests/acceptance/features/apiShareManagement/reShare.feature (hunk 2 above)
Comment (phil-davis) [reply tests/acceptance/features/apiShareManagement/reShare.feature]: copy-paste of mistake - I need to review my own code first :(

**Comment by phil-davis** [comment]
@individual-it random commented-out lines have been removed.
Ready for review again.

**Comment by individual-it** [comment]
CI failure unrelated, restarting

**Review by individual-it — APPROVED** [review]

**Comment by phil-davis** [comment]
Backport `stable10` #35547

**Pull request merged**